Assignment 5

mudgalprashant opened the pull request:
Created initial working draft for newsletter

mudgalprashant committed:
"Created initial phase of User, Email and Subscription services with JWT"

mudgalprashant committed:
Created initial working draft for newsletter

mudgalprashant committed:
Created initial working draft for newsletter

mudgalprashant committed:
fix: refactor constants

UllasCL commented on As-5/email/src/main/java/com/newsletter/email/config/KafkaConsumerConfig.java:
```diff
@@ -0,0 +1,54 @@
+  @Bean
+  ConsumerFactory<String, Email> consumerFactory() {
+    Map<String, Object> props = new HashMap<>();
+    props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
```
Don't hardcode, Pick it from env variable.

UllasCL commented on As-5/email/src/main/java/com/newsletter/email/config/KafkaProducerConfig.java:
```diff
@@ -0,0 +1,47 @@
+  @Bean
+  public ProducerFactory<String, Email> producerFactory() {
+    Map<String, Object> configProps = new HashMap<>();
+    configProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9092");
```
don't hard code, pick from env variable.

UllasCL commented on As-5/email/src/main/java/com/newsletter/email/controller/ProducerController.java:
```diff
@@ -0,0 +1,75 @@
+ */
+@AllArgsConstructor
+@RestController
+@RequestMapping("/newsletter")
```
keep this in constants.

UllasCL commented on As-5/email/src/main/java/com/newsletter/email/controller/ProducerController.java:
```diff
@@ -0,0 +1,75 @@
+   * @param token    the token
+   * @return the response entity
+   */
+  @PostMapping("/publish")
```
move it to constants

UllasCL commented on As-5/email/src/main/java/com/newsletter/email/controller/ProducerController.java:
```diff
@@ -0,0 +1,75 @@
+    if (Objects.isNull(sender)) {
+      apiResponseDto.setStatus(Constant.FAILURE_STATUS);
+      apiResponseDto.setBody(Constant.SENDER_NOT_FOUND);
+    } else if (Objects.equals(Objects.requireNonNull(userServiceClient
```
Alignment.

UllasCL commented on As-5/order/src/main/java/com/newsletter/order/controller/OrderController.java:
```diff
@@ -0,0 +1,41 @@
+ */
+@AllArgsConstructor
+@RestController
+@RequestMapping("/order")
```
move to static constants

mudgalprashant committed:
add: Included error handling and message display
add: Add authentication for creating a new Subscription
add: Added required APIs in Subscription and Subscribe
fix: Generalised api responses
fix: Fixed bug in publisher authentication
